### PR TITLE
Input Component

### PR DESCRIPTION
**Pull request by n1lsw**
# 5 I created a login component / page with a story.

### CODE REVIEW

**Comment by ReallyBoard123** [comment]
please pull the updated main , and merge it into your branch to avoid merge conflicts

File: package.json [hunk 1]
```diff
@@ -21,7 +21,9 @@
     "react": "^18.2.0",
     "react-barcode": "^1.4.6",
     "react-dom": "^18.2.0",
+    "react-icons": "^4.7.1",
     "react-router-dom": "^6.6.1",
+    "serve": "^14.1.2",
```
Comment (NoHara42) [inline package.json:26]: why is serve here

File: src/components/API.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import axios from "axios";
```
Comment (NoHara42) [inline src/components/API.tsx:1]: this should be part of a separate PR

File: package.json (hunk 1 above)
Comment (n1lsw) [reply package.json:26]: I dont know. I never installed it/touched this files. But I will remove it.

File: src/components/API.tsx (hunk 2 above)
Comment (n1lsw) [reply src/components/API.tsx:1]: Removed it

File: src/components/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+export function Button() {
+  return (
+    <div className="flex self-center">
+      <button
+        type="submit"
+        className="rounded-md h-[48px] w-[335px] bg-[#FFB43A]"
```
Comment (ReallyBoard123) [inline src/components/Button.tsx:6]: try avoiding using these [ ], and find the tailwind classes for these :)

File: src/layouts/RootLayout.tsx [hunk 4]
```diff
@@ -11,8 +11,17 @@ const routes = [
     path: "/movies",
   },
   {
+<<<<<<< HEAD
```
Comment (ReallyBoard123) [inline src/layouts/RootLayout.tsx:14]: Is this from a merge conflict in your code?

File: src/components/Input.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  // const submitfunction = async (event) => {
+  //   event.preventDefault();
+
+  //   navigate("/");
+  // };
+
+  return (
+    <form
+      id="form-placeholder"
+      className="w-96 h-[667px] bg-dark flex flex-col rounded-md"
```
Comment (ReallyBoard123) [inline src/components/Input.tsx:16]: please avoid using these absolutes and find the tailwind classes

File: src/components/Input.tsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+          your watchlist
+        </div>
+        {/* put here the button/login components created by Regina */}
+        <div className="relative">
+          <span className="absolute inset-y-0 left-0 flex items-center pl-2">
+            <button
+              type="submit"
+              className="px-4 focus:outline-none focus:shadow-outline text-body"
+            >
+              <svg
```
Comment (ReallyBoard123) [inline src/components/Input.tsx:32]: HeroIcons?

**Review by ReallyBoard123 — CHANGES_REQUESTED** [review]
So the input component looks great; just a few styling changes need to match the original design,

- [x] the text changes to white when the user will be typing in the form
- [x] the input form needs to be centered


On the other note, I see a few things in the branch that are not part of this issue, but are here to be reviewed?

**Comment by TheDoplarEffect** [comment]
Also, will look different in Storybook than in Figma file.  Have set width to full so that it addapts to screen size

**Comment by ReallyBoard123** [comment]
Who all are working on this? @n1lsw ? or @TheDoplarEffect ?

**Comment by TheDoplarEffect** [comment]
Nils started the process and I took over the final reviews last week since he was OOO

**Review by ReallyBoard123 — CHANGES_REQUESTED** [review]
So, this issue looks into creating a login page. This issue has two requirements. 
1. Button Component
2. Input Component

Currently, in this issue I noticed that input component is being used as a substitute for the login page. This issue cannot be completed before the other two issues can be closed. 

I also did not find any branch that tackles the issue for input component.

**Review by n1lsw — COMMENTED** [review]
I deleted the Button.tsx

**Review by ReallyBoard123 — CHANGES_REQUESTED** [review]
Previous comments are still not resolved.

**Review by ReallyBoard123 — APPROVED** [review]
@n1lsw 

hey, so the component that you had worked great visually, but had many nested elements. Dan showed me a way to be able to change the props in a way, that could also accept icons as props with pre-configured classNames. I thought that was a very clear solution. So I went around to update the Icon component with all the new codes, resolved the merge conflicts and updated the file. 

I have left a chunk of code in the Login page as comments, try it out and see how the new Input component works. Basically, now you can implement this in any form. Sorry about editing the code.

**Pull request merged**